flink-libraries/flink-gelly/src/test/java/org/apache/flink/graph/generator/TestUtils.java

```diff
@@ -91,8 +91,6 @@ private TestUtils() {}
 	 * @param expectedParallelism expected operator parallelism
 	 */
 	public static void verifyParallelism(ExecutionEnvironment env, int expectedParallelism) {
-		env.setParallelism(2 * expectedParallelism);
```

Comment:
Is this change needed?

Reply:
yes. before this pr, `CollectionEnvironment` override `getParallelism` method, and always return 1.  `ExecutionEnvironment#createProgramPlan` get parallelism through `ExecutionEnvironment#getParallelism`. Now, `PlanGenerator` get parallelism through `config.getParallelism()`.  When `CollectionEnvironment` is created in `ExecutionEnvironment#createCollectionsEnvironment` method, the parallelism in config will be set as 1, while after `env.setParallelism(2 * expectedParallelism);` is called, the parallelism in config was changed to 4. which is unexpected.

Reply:
I agree that the test design is a bit confusing. But doesn't this mean that this PR changes existing behavior?

Reply:
I think so. `CollectionEnvironment` is a special case that requires the parallelism is always 1.

Reply:
hi @kl0u does this make sense to you ?

Reply:
So given that users expect a specific behaviour, I think we should not change it, as this may break user code. WDYT?

Reply:
One approach is we add a constructor parameter named `defaultParallelism` for `PlanGenerator`, give the value from `ExecutionEnvironment#getParallelism`. That does not  brake current behavior, but that also means the parallelism from `ExecutionEnvironment` and from `ExecutionConfig` may be different. WDYT ?

Reply:
hi @kl0u I have reverted this change. do you have other concern ?

Reply:
@godfreyhe I agree that the propagation of the parallelism is confusing. But I think that we should not change existing behaviour in order to avoid users having unexpected results. I am having a look now at your PR.